File: app/src/test/java/app/amphora/gamesession/input/GraphemeCounterTest.java
package app.amphora.gamesession.input;

import static org.junit.Assert.assertEquals;

import org.junit.Test;

public class GraphemeCounterTest {
  @Test
  public void countsChineseCharacters() {
    assertEquals(4, GraphemeCounter.count("中文输入"));
  }

  @Test
  public void keepsSurrogateAndCombiningSequencesTogether() {
    assertEquals(1, GraphemeCounter.count("\uD83D\uDE00"));
    assertEquals(1, GraphemeCounter.count("e\u0301"));
  }

  @Test
  public void handlesEmptyInput() {
    assertEquals(0, GraphemeCounter.count(""));
    assertEquals(0, GraphemeCounter.count(null));
  }
}

File: app/src/test/kotlin/app/amphora/gamesession/input/GraphemeCounterTest.kt
package app.amphora.gamesession.input

import org.junit.Assert.assertEquals
import org.junit.Test

class GraphemeCounterTest {
    @Test
    fun countsChineseCharacters() {
        assertEquals(4, GraphemeCounter.count("中文输入"))
    }

    @Test
    fun keepsSurrogateAndCombiningSequencesTogether() {
        assertEquals(1, GraphemeCounter.count("\uD83D\uDE00"))
        assertEquals(1, GraphemeCounter.count("e\u0301"))
    }

    @Test
    fun handlesEmptyInput() {
        assertEquals(0, GraphemeCounter.count(""))
        assertEquals(0, GraphemeCounter.count(null))
    }
}
